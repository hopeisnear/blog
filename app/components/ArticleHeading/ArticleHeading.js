import React from 'react';
import { shape } from 'prop-types';
import moment from 'moment';
import { Link } from 'react-router-dom';

import imageUrlBuilder from '@sanity/image-url';
import sanity from 'utils/sanityProvider';

import './ArticleHeading.scss';

const imageBuilder = imageUrlBuilder(sanity);

function imageUrlFor(source) {
  return imageBuilder.image(source);
}

const ArticleHeading = ({ article }) => (
  <div className="article-wrapper">
    <article className="article">
      <div className="article-image">
        <Link className="article-image-link" to={`/blog/${article.slug.current}`}>
          <img
            className="content-image"
            src={imageUrlFor(article.image).ignoreImageParams().width(360).height(252)}
            alt={'article heading'}
          />
        </Link>
      </div>
      <div className="article-title-wrapper">
        <h2 className="article-title">
          <Link className="article-title-link" to={`/blog/${article.slug.current}`}>
            {article.title}
          </Link>
        </h2>
        <div className="article-timestamp">
          <time className="updated" dateTime={article.publishedAt}>{moment(article.publishedAt).format('ll')}</time> by Rafal Szczepankiewicz
        </div>
      </div>
      <article className="article-teaser">
        <div>
          <p className="article-short-description">
            {article.teaser}
          </p>
          <Link className="article-title active" to={`/blog/${article.slug.current}`}>
            Read more
          </Link>
        </div>
      </article>
    </article>
  </div>
);

ArticleHeading.propTypes = {
  article: shape({}).isRequired
};

export default ArticleHeading;

