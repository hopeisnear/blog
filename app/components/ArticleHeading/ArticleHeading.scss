@import '../../styles/variables.scss';

.article-wrapper {
  flex-basis: 100%;
  display: flex;
  margin-bottom: 35px;
  overflow: hidden;
  .article {
    display: flex;
    flex-direction: column;
  }
}

.article-image-link {
  position: relative;
  overflow: hidden;
  display: block;
  height: 140px;
}

.article {
  border: 1px solid $main-border-color;
  box-shadow: $main-box-shadow;
  margin-bottom: 30px;
  display: flex;
  flex-direction: column;
  &:hover {
    .article-image {
      img.content-image {
        opacity: 1;
        width: 115%;
      }
    }
  }
}

.article-title-wrapper {
  padding: 20px 30px 0 30px;
}

.article-teaser {
  margin: 0 30px;
  padding: 0 0 20px;
}

.article-title {
  margin-bottom: 10px;
  margin-top: 0;
}

.article-timestamp {
  font-family: $secondary-font;
  letter-spacing: $secondary-font-letter-spacing;
  font-size: $secondary-font-size;
  font-weight: $secondary-font-weight;
  line-height: $secondary-font-line-height;
  color: $secondary-color;
  margin-bottom: 10px;
  text-transform: uppercase;
}

.article-title-link {
  font-size: 28px;
  font-weight: 300;
  line-height: 32px;
}
