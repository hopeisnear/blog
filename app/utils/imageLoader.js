// eslint-disable-next-line global-require
// eslint-disable-next-line import/no-dynamic-require,global-require
export const loadImage = (imageName) => require(`images/${imageName}`);
