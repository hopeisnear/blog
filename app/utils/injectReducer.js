import React from 'react';
import { shape } from 'prop-types';
import hoistNonReactStatics from 'hoist-non-react-statics';

import getInjectors from './reducerInjectors';

export default ({ key, reducer }) => (WrappedComponent) => {
  class ReducerInjector extends React.Component {
    static contextTypes = {
      store: shape({}).isRequired
    };

    injectors = getInjectors(this.context.store);

    componentWillMount() {
      const { injectReducer } = this.injectors;

      injectReducer(key, reducer);
    }

    render() {
      return <WrappedComponent {...this.props} />;
    }
  }

  return hoistNonReactStatics(ReducerInjector, WrappedComponent);
};
