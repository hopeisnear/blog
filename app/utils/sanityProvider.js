import sanityClient from '@sanity/client';

export default sanityClient({
  projectId: 'klyyehyz',
  dataset: 'production',
  token: 'skuMGzswNGLOzsNVtmsc6RAV1IoNP3rH665qW1gNdJ9SkZJFeWM1mqJQZ8eMbjTUnZLN9mf3X1nS0IhKxvorzQABMr9cab3XmLG1rPjJQUzcEG1NPVuWhUprGMBukGiCtefng8aRvLXyjA3DoB5hekZw0Hfses0A0V9UjjA1sBAYXE9ZRUKw',
  useCdn: false
});
