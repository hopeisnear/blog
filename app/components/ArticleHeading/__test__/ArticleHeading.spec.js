import React from 'react';
import { shallow } from 'enzyme';
import { stub } from 'sinon';
import * as imageLoader from 'utils/imageLoader';
import ArticleHeading from '../ArticleHeading';

describe('<ArticleHeading />', () => {
  test('should render <ArticleHeading />', () => {
    stub(imageLoader, 'imageUrlFor').callsFake(image => ({ ignoreImageParams: () => ({ width: () => ({ height: () => image }) }) }));
    const article = { slug: { current: 'article-1' }, title: 'article-1-title', image: 'image1', publishedAt: '2018-08-18T15:01:53Z' };

    const articleHeadingComponent = shallow(<ArticleHeading article={article} />);

    expect(articleHeadingComponent).toMatchSnapshot();
  });

  afterEach(() => {
    imageLoader.imageUrlFor.restore();
  });
});
