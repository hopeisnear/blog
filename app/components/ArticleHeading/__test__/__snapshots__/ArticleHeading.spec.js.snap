// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleHeading /> should render <ArticleHeading /> 1`] = `
<article
  className="ArticleHeading"
>
  <div
    className="article__header"
  >
    <Link
      className="header-image__link"
      replace={false}
      to="/article-1"
    >
      <img
        alt="article"
        className="header-image"
        src="image1"
      />
    </Link>
  </div>
  <div
    className="article__title"
  >
    <h2
      className="title__heading"
    >
      <Link
        className="title__link"
        replace={false}
        to="/article-1"
      >
        article-1-title
      </Link>
    </h2>
    <div
      className="title__timestamp"
    >
      <time
        dateTime="2018-08-18T15:01:53Z"
      >
        Aug 18, 2018
      </time>
      <span>
         by Rafal Szczepankiewicz
      </span>
    </div>
  </div>
  <article
    className="article__body"
  >
    <div>
      <p />
      <Link
        className="active"
        replace={false}
        to="/article-1"
      >
        <span>
          Read more Â»
        </span>
      </Link>
    </div>
  </article>
</article>
`;
